File: 2384-root-equals-sum-of-children/2384-root-equals-sum-of-children.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    bool checkTree(TreeNode* root) {

        if (root == NULL){
            return false;
        }

        long total = 0;

        if (root->left != NULL){
            total += root->left->val;
        }
        if(root->right != NULL){
            total += root->right->val;
        }

        if (root->val == total){
            return true;
        }

        return false;
        
    }
};

File: 2384-root-equals-sum-of-children/2384-root-equals-sum-of-children.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def checkTree(self, root: Optional[TreeNode]) -> bool:
        
        if not root:
            return False

        total = 0
        if root.left:
            total += root.left.val
        if root.right:
            total += root.right.val

        if total == root.val:
            return True
        return False